src/actors/symbolication.rs

```diff
@@ -452,44 +489,54 @@ impl SourceLookup {
         Some((pre_context, context, post_context))
     }
 
-    fn get_object_index_by_addr(&self, addr: u64) -> Option<usize> {
-        for (i, (info, _)) in self.inner.iter().enumerate() {
-            let start_addr = info.raw.image_addr.0;
+    fn get_object_index_by_addr(&self, addr: u64, addr_mode: AddrMode) -> Option<usize> {
+        for &(module_index, ref info, _) in self.inner.iter() {
+            match addr_mode {
```

Comment:
_nit_: I'd turn this around and do the match first. The `ModRel` case then becomes a simple `.find()` on `inner`.